packages/components/data-table/src/cell.js

```diff
@@ -114,14 +113,14 @@ DataCell.propTypes = {
   isCondensed: PropTypes.bool,
   isTruncated: PropTypes.bool,
   shouldIgnoreRowClick: PropTypes.bool,
-  handleRowCollapseClick: requiredIf(
-    PropTypes.func,
-    (props) => props.isTruncated
-  ),
   shouldRenderCollapseButton: requiredIf(
     PropTypes.bool,
     (props) => props.isTruncated
   ),
+  handleRowCollapseClick: requiredIf(
+    PropTypes.func,
+    (props) => props.shouldRenderCollapseButton
+  ),
   isRowCollapsed: requiredIf(PropTypes.bool, (props) => props.isTruncated),
```

Comment:
Here it should also be dependent on the `shouldRenderCollapseButton` prop and not `isTruncated`, right?
```suggestion
  isRowCollapsed: requiredIf(PropTypes.bool, (props) => props. shouldRenderCollapseButton),
```